src/lsp/main.ts

```diff
@@ -658,6 +663,33 @@ export async function getClojuredocs(
     }
 }
 
+// TODO: This feels a bit brute, what are other ways to wait for the client to initialize?
```

Comment:
This is fine as-is but we could pull in a package for this -- like https://github.com/devlato/async-wait-until -- to get something with behavior that's easier to wrap your head around.

Reply:
@PEZ If you return `onReadyPromise` from `startClient` you can just await `startClient` to know when the client has started and is ready, rather than using this `getClient` function. I could help with that if you want.

Reply:
Oh, that would be less brute. I'll have a go at it and see where it brings me. Thanks!

Reply:
No problem! You may want to put the code that was previously after the await within `startClient` inside a .then.

So it would be like

```typescript
return onReadyPomise.then(_ => { ...do-stuff });
```

That should return the promise as well as run the code we want to run after the client starts.